libs/Microsoft.MixedReality.WebRTC.Native/src/peer_connection.cpp

```diff
@@ -1698,11 +1730,22 @@ void PeerConnectionImpl::SynchronizeTransceiversUnifiedPlan(bool remote) {
   // for the ones without one yet.
   RTC_DCHECK_GE(rtp_transceivers.size(), wrappers.size());
   auto it_wrapper = wrappers.begin();
+  RTC_LOG(LS_INFO) << "Synchronizing " << rtp_transceivers.size()
+                   << " RTP transceivers with " << wrappers.size()
+                   << " transceiver wrappers (remote = " << remote << ").";
   for (auto&& rtp_tr : rtp_transceivers) {
     const int mline_index = ExtractMlineIndexFromRtpTransceiver(rtp_tr);
     // Create a wrapper if it doesn't exist yet
-    if (it_wrapper == wrappers.end()) {
+    if ((it_wrapper == wrappers.end()) || ((*it_wrapper)->impl() != rtp_tr)) {
```

Comment:
What does it mean here if there is another transceiver already?

Reply:
The transceivers are sorted by address in memory. You have 2 arrays being iterated in parallel. If one value doesn't match with the other, it means one array (`wrappers`) is missing an element and you need to insert it.
```
RTP | Wrapper
----|--------
A   | A
B   | B   v
C   | D   ^ Need to insert here because C != D
D   |
```

Reply:
Yeah it's reasonably clear once you un-hide the surrounding lines 😄 
nit: might be worth adding a comment for good measure.

Reply:
Yes the comment confuses the reader I think, but thinking about the actual algorithm (sort 2 arrays to avoid O(n^2) comparison) makes things clear.

Reply:
Fixed.